source/core/src/test/com/csse3200/game/inventory/items/ConsumableItemTest.java

```diff
@@ -0,0 +1,72 @@
+package com.csse3200.game.inventory.items;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.internal.matchers.text.ValuePrinter.print;
+
+import com.csse3200.game.extensions.GameExtension;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(GameExtension.class)
+public class ConsumableItemTest {
+    private TestableItem1 item1;
+    private TestableItem1 item2;
+    private static ItemUsageContext context;
+
+    /**
+     * Concrete subclass of ConsumableItems for testing purposes.
+     */
+    private static class TestableItem1 extends ConsumableItem {
+        public int numUsed = 0;
+
+        public TestableItem1(String name, int itemCode, int limit, int quantity) {
+            super(name, itemCode, limit, quantity);
+        }
+
+        @Override
+        public void useItem(ItemUsageContext inputs) {
+            //for testing
+            if (super.isEmpty()) {
+                throw new ArrayIndexOutOfBoundsException("This item has ran out of uses");
+            }
+            else {
+                this.quantity--;
+                super.isEmpty();
+            }
+        }
+    }
+
+    @BeforeEach
+    void setUp() { // Initialize TestableItem and ItemUsageContext
+        item1 = new TestableItem1("Test1", 3, 10, 3);
```

Comment:
create a new ConsumableItem to test

Reply:
You cannot create a ConsumableItem - this is an abstract class.